plugins/modules/rocketchat.py

```diff
@@ -100,6 +100,11 @@
     elements: dict
     description:
       - Define a list of attachments.
+  option_is_pre740:
+    description:
+      - If V(true), the payload matches Rocket Chat prior to 7.4.0 format
+    type: bool
+    default: true
```

Comment:
Couple of mandatory things:
```suggestion
  option_is_pre740:
    description:
      - If V(true), the payload matches Rocket Chat prior to 7.4.0 format.
    type: bool
    default: true
    version_added: 10.5.0
```
But then, calling an option "option_XYZ" sounds odd to my ears. Some suggestions (feel free to invent something different or take your pick):
- `pre740_payload`
- `is_pre740`
- `payload_type` (not a boolean the, but `str` with `choices: [ pre740, version740 ]` - certainly better option names can be found here)
- `feature_pre740`

Just suggestions, no need to pick them literally. ;-)